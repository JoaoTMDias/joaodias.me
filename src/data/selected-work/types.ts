/**
 * This file is open-source. This means that it can be reproduced in whole
 * or in part, stored in a retrieval system transmitted in any form, or by
 * any means electronic with my prior permission as an author and owner
 * Please refer to the terms of the license agreement in the root of the project
 *
 * (c) 2022 joaodias.me, Rights Reserved.
 */

export interface SelectedProjectsProps {
	data?: Project[];
}

export interface Project {
	id?: string;
	title?: string;
	shortDescription?: string;
	thumbnail?: Thumbnail;
	skills?: string[];
	details?: Details;
}

export interface Details {
	date?: string;
	description?: string;
	cover?: Thumbnail;
	sourceCode?: string;
	problem?: string[];
	solution?: string[];
	photos?: Thumbnail[];
}

export interface Thumbnail {
	src?: string;
	width?: number;
	height?: number;
	alt?: Alt;
}

export enum Alt {
	Cenas = "cenas",
}
