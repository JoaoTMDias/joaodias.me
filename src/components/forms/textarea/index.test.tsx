import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { TextAreaInput, defaultProps } from "./index";
import { ITextAreaInputProps } from "../types";

afterEach(cleanup);

let props: ITextAreaInputProps;

describe("<TextAreaInput />", () => {
	beforeEach(() => {
		props = {
			...defaultProps,
			value: "",
		};
	});
	it("should render without errors", () => {
		const component = render(<TextAreaInput {...props} />);

		expect(component).toMatchSnapshot();
	});

	it("should call onChange when typing", async () => {
		const onChangeMock = jest.fn();
		props = {
			...props,
			onChange: onChangeMock,
		};
		const { findByTestId } = render(<TextAreaInput {...props} />);

		const input = await findByTestId("component-text-area-input");

		fireEvent.change(input, {
			target: {
				value: "teste",
			},
		});

		expect(onChangeMock).toHaveBeenCalled();
	});

	it("should have isFocused class when focused", async () => {
		const { findByTestId } = render(<TextAreaInput {...props} />);

		const input = await findByTestId("component-text-area-input");
		const wrapper = await findByTestId("component-text-area-wrapper");

		fireEvent.focus(input);

		expect(wrapper).toHaveClass("isFocused");
	});

	it("should not have isFocused class when loses focus", async () => {
		const { findByTestId } = render(<TextAreaInput {...props} />);

		const input = await findByTestId("component-text-area-input");
		const wrapper = await findByTestId("component-text-area-wrapper");

		fireEvent.blur(input);

		expect(wrapper).not.toHaveClass("isFocused");
	});

	it("should have helper text", async () => {
		const helperText = "This a helper text";
		props = {
			...props,
			helperText,
		};
		const { findByTestId } = render(<TextAreaInput {...props} />);

		const helper = await findByTestId("component-text-area-helper");

		expect(helper).toHaveTextContent(helperText);
	});
});
