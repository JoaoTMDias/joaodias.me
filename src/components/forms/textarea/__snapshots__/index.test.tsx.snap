// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextAreaInput /> should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm sc-AxirZ kOZIao"
        data-form="textarea"
        data-testid="component-text-area-wrapper"
      >
        <label
          class="content"
          data-testid="component-text-area-label"
          for="text-area-input"
        >
          <span
            class="label"
          />
          <textarea
            class="input undefined"
            cols="500"
            data-testid="component-text-area-input"
            id="text-area-input"
            maxlength="500"
            minlength="10"
            name="text-area-input"
            placeholder="Text Area placeholder..."
          />
          
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm sc-AxirZ kOZIao"
      data-form="textarea"
      data-testid="component-text-area-wrapper"
    >
      <label
        class="content"
        data-testid="component-text-area-label"
        for="text-area-input"
      >
        <span
          class="label"
        />
        <textarea
          class="input undefined"
          cols="500"
          data-testid="component-text-area-input"
          id="text-area-input"
          maxlength="500"
          minlength="10"
          name="text-area-input"
          placeholder="Text Area placeholder..."
        />
        
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
